students/arkadiusz_kasprzyk/lesson_05_lists/chess_board.py

```diff
@@ -0,0 +1,57 @@
+def rep(x, length=None, times=None):
+    """
+    Replicates x (list or tuple) to the given length or times many.
```

Comment:
The idea is to write imperative and use statements as names in unit tests. So: `Replicate a sequence to fill length`. And then in `test_abc_is_replicated_3_times`.

Reply:
I don't quite get it... 
* unit tests cannot serve as documentation (they are not part of a file/source code and may be missed), 
* documentation should be maximaly helpful for the user (for newbies too!), while phrases like "Replicate a sequence to fill length" sounds somewhat enigmatic (although very professional!)...
* ... in a sense that, strictly speeking, replicating n times (final length unknown and not an issue) is not the same as filling to the given length.
So imperative is not well defined, and maybe such functions should not be written, but they are useful (I have copied it from basic R). This is matter for a long discussion, again!, how imperatives should be stated, and should they always be atomic (in what sense?). Convienience vs clarity... I don't know! 
Sorry for this! Maybe I just don't understatand what you mean.

Reply:
I wasn't picking so much on the formulation (mine is probably worse) as on the grammatical form (fills vs fill). The idea is, don't be so sure what a complicated piece of code is doing, so defensively you should describe it with your wish "replicate string to a given length" instead "replicates strings to a given length".

On the other hand with tests you want to say what the positive result of the test means and try to be specific "joins a strings from an iterable", "joins a strings from a list", "raises an exception on non-iterable" etc.

Unit tests serve as documentation of your non-obvious intentions on edge cases. They are also very useful, when trying to use the code. They show an example of such usage. Also "the documentation" is automatic and will scream at you, if you have misunderstood and changed the intention. I'm always disappointed when reading manual page and `/example` doesn't bring up EXAMPLES section. It means I have to read the boring stuff.

Reply:
OK, now I got it. It's one of the cases when single 's' has such a deep meaning... 
However I feel that we do not understatnd *documentation* in the same way: for me it is something what serves a final user who has no access to tests (but only to the final product), whereas for you (I guess) it's sth that serves a developer who has access to the whole history of the project (hence to tests too). 
After all, it's very unclear to me how to write docstrings and whom it should serve best. I've read some PEPs, heve seen few conventions in action, PyCharm suggests another, so... ??? Of course there're always sections like *In*, *Out*, *Examples*, *Description* but theirs format and content is rather not precised.